src/Action/SetObjectFieldValueAction.php

```diff
@@ -121,10 +120,13 @@ public function __invoke(Request $request): JsonResponse
         }
 
         $propertyPath = new PropertyPath($field);
+        $rootObject = $object;
 
         // If property path has more than 1 element, take the last object in order to validate it
         if ($propertyPath->getLength() > 1) {
-            $object = $this->propertyAccessor->getValue($object, $propertyPath->getParent());
+            $parent = $propertyPath->getParent();
+            \assert(null !== $parent);
```

Comment:
What about adding this to the previous if?

Reply:
The `getParent` method is 
```
public function getParent()
    {
        if ($this->length <= 1) {
            return null;
        }

        $parent = clone $this;

        --$parent->length;
        $parent->pathAsString = substr($parent->pathAsString, 0, max(strrpos($parent->pathAsString, '.'), strrpos($parent->pathAsString, '[')));
        array_pop($parent->elements);
        array_pop($parent->isIndex);

        return $parent;
    }
```
So checking for both length and parent is redundant. 

We may check for parent instead of length ?

Reply:
Done